hypothesis-python/src/hypothesis/internal/conjecture/utils.py

```diff
@@ -442,3 +444,23 @@ def reject(self):
                 self.data.mark_invalid()
             else:
                 self.force_stop = True
+
+    @staticmethod
+    def _calc_stopping_value(average_size, max_size):
+        stop_val = 1 - 1.0 / (1 + average_size)
+        if stop_val == 0 or max_size == float("inf"):
+            return stop_val
+        # For small max_size, the infinite-series stop_val is a poor approximation,
+        # and while we can't solve the polynomial a few rounds of iteration gets us
+        # a very good approximate solution in almost all cases.
+        for _ in range(5):
+            est_mean = _stop_val_to_avg_size(stop_val, max_size)
+            inc = 0.5 * (average_size - est_mean) / (est_mean + average_size)
```

Comment:
Where does this come from? This can have `stop_val * (1 + inc) > 1` (e.g. if `max_size = 101, average_size=100, inc=0.112`).

Reply:
If `average_size << max_size`, then our `_stop_val_to_avg_size` function is approximately linear and this is a quick and effective update.  If it's *not*, then we ensure that `stop_val < 1` by the repeated halving below - this also avoids oscillation by ensuring that `est_mean <= average_size`.

I'll add a comment and rename some variables for clarity - `stopping_value` is properly `p_continue`!